pages/index.js

```diff
@@ -1,68 +1,29 @@
-import envFlag from 'node-env-flag';
 import React from 'react';
-import Meta from '../frontend/components/meta';
-import Header from '../frontend/components/header';
-import SuggestionAndSearch from '../frontend/components/suggestion-and-search';
 import { BadgeExamples } from '../frontend/components/badge-examples';
-import MarkupModal from '../frontend/components/markup-modal';
-import Usage from '../frontend/components/usage';
-import Footer from '../frontend/components/footer';
-import badgeExampleData from '../badge-examples.json';
-import { prepareExamples, predicateFromQuery } from '../frontend/lib/prepare-examples';
+import ExamplesPage from '../frontend/components/examples-page';
+import { baseUri, longCache } from '../frontend/constants';
 
-const baseUri = process.env.BASE_URL;
-const longCache = envFlag(process.env.LONG_CACHE, false);
+export default class IndexPage extends ExamplesPage {
 
-export default class IndexPage extends React.Component {
-  state = {
-    query: null,
-    example: null
-  };
-
-  constructor(props) {
-    super(props);
-    this.preparedExamples = prepareExamples(
-      badgeExampleData,
-      () => predicateFromQuery(this.state.query));
+  getBody() {
+    if ((this.state.query == null) || (this.state.query.length === 0)) {
+      return this.preparedExamples.map(category => (
+        <a href={'/examples/' + category.category.id}>
+          <h3 id={category.category.id}>{ category.category.name }</h3>
+        </a>
+      ));
+    } else if (this.state.query.length === 1) {
+      return (
+        <div>Search term must have 2 or more characters</div>
+      );
+    } else {
+      return(<BadgeExamples
```

Comment:
Would it be worth adding a ~500ms delay here?
It may help reduce the amount of badges loaded.
something like:
```javascript
var searchTimeout = 0;
...
clearTimeout(searchTimeout);
searchTimeout = setTimeout(runSearch, 500);
```